### PR TITLE
Documentation for parsing user defined units when reading into QTable.

### LINKED ISSUE #8883: User defined units fail to parse when reading into QTable
I construct a QTable that takes in data as a user defined unit and output it to the current directory. 
```
bandpass_sol_lum = u.def_unit('bandpass_sol_lum') 
lum_array = np.ones((5,5)) * bandpass_sol_lum

qt = QTable()
qt['Luminosity'] = QTable.Column(lum_array) 
qt.write('output_lum.fits',overwrite=True,format='fits')
```
When I read it back in: 
```
qt_readIn = QTable()
qt_readIn['Luminosity'] = QTable.read('output_lum.fits',format='fits') # <--Exception thrown
```
I get:
```
YamlParseError: 'bandpass_sol_lum' did not parse as unit: At col 0, bandpass_sol_lum is not a valid unit. 
```
Is it possible at all to parse user defined units when reading them back in from a FITS file?
@zpace

My environment:
macOS 10.14.3
python 3.6.8
astropy 3.2
numpy 1.16.4

### PR DESCRIPTION
**Pull request by JasonS09**
### Description

This PR addresses requested documentation on the solution for issue #8883 

Added a new example in _Combining and Defining Units_ based on the solution posted for the issue. I also added a couple of reference labels to be used in a new section for tables called _Tips and Tricks_, which also contains an example of how to handle the issue.

Fixes #8883 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/table/tips_and_tricks.rst
Comment (pllim) [inline docs/table/tips_and_tricks.rst:14]: What is "titter"?

**Review by pllim — COMMENTED** [review]
This might have to wait till after #12011 goes in.

**Comment by pllim** [comment]
Also why is there a merge commit?

**Comment by pllim** [comment]
p.s. Also please don't skip CI, because we want the doctest to run. We can wait till the next push for this though.

File: docs/table/tips_and_tricks.rst
Comment (JasonS09) [reply docs/table/tips_and_tricks.rst:14]: I took one of the arbitrary defined units shown in a previous example and adapted it to this one.

Comment (pllim) [reply docs/table/tips_and_tricks.rst:14]: For official doc, would be less confusing if we stick to something closer to astronomy... 💭

**Comment by JasonS09** [comment]
> Also why is there a merge commit?

Oh, now I know. I had to delete my local copy of Astropy and set up a new clone. Maybe that commit has to do with it and sneaked into this PR.

File: docs/table/tips_and_tricks.rst
Comment (JasonS09) [reply docs/table/tips_and_tricks.rst:14]: Yeah, makes sense. Gonna do those changes. Maybe I should just use something like "my_unit"?

And what about the CI exceptions? Should I ignore them?

**Comment by pllim** [comment]
`NameError: name 'np' is not defined` -- I think you need to add a `import numpy as np` in your example.

**Comment by JasonS09** [comment]
@pllim Can you help me here? 

`UNEXPECTED EXCEPTION: AstropyUserWarning("The unit 'my_unit' could not be saved in native FITS format and hence will be lost to non-astropy fits readers. Within astropy, the unit can roundtrip using QTable, though one has to enable the unit before reading.")`

The warning appears even when I'm testing it manually with the example provided in the issue. Not sure if I should handle this error since it describes what I'm documenting here.

**Comment by pllim** [comment]
I think the warning is legit... If there is no way to modify your example so it does not emit the warning, you can try use https://github.com/astropy/pytest-doctestplus#ignoring-warnings

**Comment by pllim** [comment]
p.s. There is also a conflict, so please *rebase*, thank you!

File: docs/table/index.rst
Comment (eerovaher) [inline docs/table/index.rst:373]: I'm not sure a separate tips and tricks page is the best place for the new documentation. Wouldn't https://docs.astropy.org/en/latest/table/mixin_columns.html#quantity-and-qtable or https://docs.astropy.org/en/latest/table/io.html be more appropriate?

File: docs/table/tips_and_tricks.rst
Comment (eerovaher) [inline docs/table/tips_and_tricks.rst:7]:
```suggestion
A |QTable| can be used to store data as :ref:`user-defined
```
Not all of the existing documentation has been updated to take advantage of the [available ReST substitutions](https://github.com/astropy/astropy/blob/main/docs/common_links.txt), but new documentation should use them.

File: docs/table/tips_and_tricks.rst
Comment (eerovaher) [inline docs/table/tips_and_tricks.rst:19]: I find using `np.ones` here a bit distracting.

File: docs/table/tips_and_tricks.rst
Comment (eerovaher) [inline docs/table/tips_and_tricks.rst:21]:
```suggestion
  >>> qt['speeds'] = np.arange(5) * lps
```
Isn't this simpler than using the intermediate variable `lps_array`?

File: docs/table/tips_and_tricks.rst
Comment (eerovaher) [inline docs/table/tips_and_tricks.rst:28]: There should first be an example of what happens if the table is read without enabling units. If anyone ever needs to look up this documentation then the presence of the exact error message might make it easier to find. Furthermore, if `astropy` is ever updated so that the explicit enabling of new units is no longer necessary then it would be good if that test fails so that updating the docs wouldn't be forgotten.

File: docs/units/combining_and_defining.rst
Comment (eerovaher) [inline docs/units/combining_and_defining.rst:54]:
```suggestion
```
These lines are duplicating what is already present.

File: docs/units/combining_and_defining.rst
Comment (eerovaher) [inline docs/units/combining_and_defining.rst:59]: If you wish this example to have its own header it should be a subsection in the Defining Units section and not a separate section.

File: docs/units/combining_and_defining.rst
Comment (eerovaher) [inline docs/units/combining_and_defining.rst:115]: I'm not sure if there's much point in duplicating this example. It might be a good idea to only have the example written out in the Table documentation and to just link to that from here.

**Review by eerovaher — CHANGES_REQUESTED** [review]

**Comment by JasonS09** [comment]
@eerovaher I made the changes you requested. One thing though, I could not find a way to put the warning as is in the example as the doctests were looking for the exception that causes it instead, so I think in the end it's not fully illustrative... I don't know what do you think, or if you do know one way to show the warning message without having to write the exception in the example.

File: docs/table/mixin_columns.rst
Comment (eerovaher) [inline docs/table/mixin_columns.rst:171]: The `+SHOW_WARNINGS` directive should make the warning appear in the doctest., and saving the table in a variable is probably more representative of how `QTable.read()` is typically used.
```suggestion
  >>> qt4 = QTable.read('Lps_output.fits', format='fits')  # doctest: +SHOW_WARNINGS
```

File: docs/units/combining_and_defining.rst
Comment (eerovaher) [inline docs/units/combining_and_defining.rst:55]: There shouldn't be any links that are not used anywhere.
```suggestion
```

File: docs/units/combining_and_defining.rst
Comment (eerovaher) [inline docs/units/combining_and_defining.rst:103]: Why is this line marked as having changed?

**Review by eerovaher — CHANGES_REQUESTED** [review]
I did recommend not having two identical examples in two different places in the documentation, but it might be good idea to mention this problem in `docs/units/combining_and_defining.rst` and to add a link to the example in the Table documentation there.

I'm not sure if `docs/table/mixin_columns.rst` is the best place for the example, `docs/table/io.rst` also looks like a good location. Perhaps a maintainer can weigh in on this.

File: docs/units/combining_and_defining.rst
Comment (JasonS09) [reply docs/units/combining_and_defining.rst:55]: I added it because I'm using it in the `docs/table/mixin_columns.rst` new example. Although, with a naked eye, I'm unsure as to why is it being marked as never used in the editor, but it's working in my built html files.

Comment (eerovaher) [reply docs/units/combining_and_defining.rst:55]: It is used, I was mistaken.

File: docs/units/combining_and_defining.rst
Comment (JasonS09) [reply docs/units/combining_and_defining.rst:103]: I suppose that as I am adding a new line for the link, there is one new line added in the end of the file and that's what is being marked.

File: docs/units/combining_and_defining.rst
Comment (eerovaher) [reply docs/units/combining_and_defining.rst:55]: But if you want to link to a section then the link needs to be above the section title. If it's below it, like it is now, then it's actually linking to an unmarked location just below the title. You might not be able to tell the difference because this section is close to the bottom of the page, but if you make the browser window very narrow before clicking on the link you should see that the link takes you past the title.

**Comment by eerovaher** [comment]
The documentation being added here describes an I/O problem, so I would say `docs/table/io.rst` would be a better location, but `docs/table/mixin_columns.rst` should mention the section in `io.rst` and provide a link to it.

**Comment by JasonS09** [comment]
@eerovaher Hello! It seems there is an issue with the Read the Docs build but I don't really understand what is this about. Can you help me here?

**Comment by eerovaher** [comment]
I'm seeing a bunch of warnings like this:
```
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/12016/docs/development/codeguide.rst:188: WARNING: py:func reference target not found: super
```
These are not caused by this pull request and have been fixed by #12235. The documentation build should succeed if you rebase.

**Comment by eerovaher** [comment]
I should have ran the example that this pull request would add to the documentation much sooner, but now when I finally did it I received the following warning:

WARNING: UnitsWarning: 'Lps' did not parse as unit: At col 0, Lps is not a valid unit. Did you mean PS, Ps, pS or ps? If this is meant to be a custom unit, define it with 'u.def_unit'. To have it recognized inside a file reader or other code, enable it with 'u.add_enabled_units'. For details, see https://docs.astropy.org/en/latest/units/combining_and_defining.html [astropy.units.core]

To me it looks like the warning message added by d965bcdc9b291edee22703840bf9ef74945d691b has already resolved #8883 and the documentation updates might not be needed. @mhvk, what do you think?

**Comment by pllim** [comment]
It is unfortunate that this got stale. Was it close and was just waiting for a minor response? Maybe one of the core maintainers can wrap this up if there is still interest and the original author is no longer interested to revisit?

**Comment by mhvk** [comment]
Had a quick look and it seems it was close. @eerovaher - would you be able to have a look and see whether your comments were addressed?

File: docs/table/io.rst
Comment (eerovaher) [inline docs/table/io.rst:84]: The actual message should be here.

File: docs/units/combining_and_defining.rst
Comment (eerovaher) [inline docs/units/combining_and_defining.rst:104]: This too mentions FITS explicitly.

**Pull request closed without merging**
